modules/katamari/src/main/ts/ephox/katamari/api/Type.ts

```diff
@@ -20,12 +20,34 @@ const isSimpleType = <Yolo>(type: string) => (value: any): value is Yolo =>
 const eq = <T> (t: T) => (a: any): a is T =>
   t === a;
 
+const getPrototypeOf = Object.getPrototypeOf;
+
 export const isString: (value: any) => value is string =
   isType('string');
 
 export const isObject: (value: any) => value is Object =
   isType('object');
 
+export const isPlainObject = (value: unknown): value is Object => {
+  if (!isObject(value)) {
+    return false;
+  }
+
+  const constructor = value.constructor?.name;
```

Comment:
Should this type of thing be pulled out into a reusable function, as we're already doing this type of check with `String` and `Array` in `typeOf`? This does cover more cases as well, which are likely not working on IE 11.

Reply:
Actually, I was just thinking about this more... maybe IE keeps the same references across windows, as otherwise the `String` and `Array` tests should be failing. If that's the case, then we can likely skip the to string checks below :shrug: 

Either way, given this is being used multiple times and across multiple functions I definitely think we should be looking to pull this out into a separate function.

Reply:
I've done an experiment related to this, and nowI want to see if the tests still pass in all browsers / OSes with this new thing. Either I'll pull the checks into a separate function or I'll remove them, depending on how Jenkins feels